content/4_Gold/Knapsack_DP.mdx

```diff
@@ -94,16 +96,18 @@ $\texttt{dp}[x]$ if $x < 0$.
 
 using namespace std;
 
-typedef long long ll;
+using ll = long long;
+
+ll dp[1000001];
 
 int main() {
 	int n;
 	cin >> n;
-	ll dp[n+1]{}; dp[0] = 1;
-	for (int i = 1; i<= n; i++) {
-		for (int j = 1; j <= 6; j++) {
-			if (i-j >= 0) {
-				dp[i] += dp[i-j];
+	dp[0] = 1;
+	for(int i = 1; i <= n; i++) {
+		for(int j = 1; j <= 6; j++) {
+			if(i - j >= 0) {
+				dp[i] += dp[i - j];
 				dp[i] %= 1000000007;
 			}
 		}
```

Comment:
two options
1) change `ll` to `int` (long longs are unnecessary)
2) move `dp[i] %= 1000000007;` outside `for(int j = 1; j <= 6; j++) {` (the `%` operation takes a lot of time, so it would be better to move it outside the loop; won't overflow bc `ll`)

also change java sol too

Reply:
not really sure if `%` is slow in java